### PR TITLE
Enhancement/2507 remove module datapoints for individual settings

### PR DESCRIPTION
**Pull request by eugene-manuilov**
## Summary

Addresses issue #2507

## Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: includes/Modules/Analytics.php [hunk 1]
```diff
@@ -950,7 +795,8 @@ function ( Google_Service_Analytics_Account $account ) {
 					);
 				} else {
 					// Get the account ID from the saved settings - returns WP_Error if not set.
```
Comment (felixarntz) [inline includes/Modules/Analytics.php:797]:
```suggestion
					// Get the account ID from the saved settings.
```

File: includes/Modules/Tag_Manager.php [hunk 2]
```diff
@@ -160,6 +155,7 @@ function( $container_id ) {
 				return is_wp_error( $container_id ) || ! $container_id;
```
Comment (felixarntz) [inline includes/Modules/Tag_Manager.php:155]: See above:
```suggestion
				return ! $container_id;
```

**Review by felixarntz — CHANGES_REQUESTED** [review]
Mostly LGTM. There are a few minor things to clean up related to unnecessary `is_wp_error` checks mostly.

File: includes/Modules/AdSense.php [hunk 3]
```diff
@@ -538,9 +417,10 @@ protected function parse_data_response( Data_Request $data, $response ) {
...
 				if ( $data['maybeSetAccount'] && ! empty( $accounts ) ) {
-					$account_id = $this->get_data( 'account-id' );
-					if ( is_wp_error( $account_id ) || ! $account_id ) {
-						$this->set_data( 'account-id', array( 'accountID' => $accounts[0]->id ) );
+					$option     = $this->get_settings()->get();
+					$account_id = $option['accountID'];
+					if ( empty( $account_id ) && ! empty( $accounts[0]->id ) ) {
+						$this->get_settings()->merge( array( 'accountID' => $accounts[0]->id ) );
 					}
 				}
```
[the remark this answers is not in the document]
Comment (eugene-manuilov) [reply includes/Modules/AdSense.php]: Ok, removed it.

File: includes/Modules/AdSense.php [hunk 4]
```diff
@@ -680,7 +560,8 @@ protected function create_adsense_earning_data_request( array $args = array() )
 			)
 		);
 
-		$account_id = $this->get_data( 'account-id' );
+		$option     = $this->get_settings()->get();
+		$account_id = $option['accountID'];
 		if ( is_wp_error( $account_id ) ) {
 			return $account_id;
 		}
```
[the remark this answers is not in the document]
Comment (eugene-manuilov) [reply includes/Modules/AdSense.php]: Updated!

**Review by felixarntz — APPROVED** [review]
Great!

**Comment by aaemnnosttv** [comment]
more-red-than-green

**Pull request merged**